Core/SubscriptionManager/src/main/java/eu/neclab/ngsildbroker/subscriptionmanager/service/NotificationHandlerMQTT.java

```diff
@@ -58,7 +64,14 @@ private MqttClient getClient(URI callback, Map<String, String> clientSettings) {
 		URI baseURI = URI.create(callback.getScheme() + "://" + callback.getAuthority());
 		MqttClient result = uri2client.get(baseURI);
 		if (result == null) {
-			String mqttVersion = clientSettings.get(NGSIConstants.MQTT_VERSION);
+			String mqttVersion = null;
+			if(clientSettings != null) {
+				mqttVersion = clientSettings.get(NGSIConstants.MQTT_VERSION);
+				
+			} else {
+				mqttVersion = NGSIConstants.DEFAULT_MQTT_VERSION; 
+			}
+			
 			int port = callback.getPort();
 			if (port == -1) {
 				port = 1883;
```

Comment:
[Query] the default MQTT port is 1883 for MQTT over TCP and 8883 for mqtts. Are we taking care of MQTTS